openreview/conference/builder.py

```diff
@@ -0,0 +1,400 @@
+
+    def set_id(self, id):
+        self.id = id
+
+    def get_id(self):
+        return self.id
+
+    def set_name(self, name):
+        self.name = name
+
+    def get_name(self):
+        return self.name
+
+    def set_short_name(self, name):
+        self.short_name = name
+
+    def get_short_name(self):
+        return self.short_name
+
+    def set_reviewers_name(self, name):
+        self.reviewers_name = name
+
+    def set_program_chairs_name(self, name):
+        self.program_chairs_name = name
+
+    def get_program_chairs_id(self):
+        return self.id + '/' + self.program_chairs_name
+
+    def get_reviewers_id(self):
+        return self.id + '/' + self.reviewers_name
+
+    def set_type(self, type):
+        self.type = type
+
+    def get_type(self):
+        return self.type
+
+    def set_conference_groups(self, groups):
+        self.groups = groups
+
+    def get_conference_groups(self):
+        return self.groups
+
+    def set_homepage_header(self, header):
+        self.header = header
+
+    def get_homepage_options(self):
+        options = {}
+        if self.name:
+            options['subtitle'] = self.name
+        if self.header:
+            options['title'] = self.header.get('title')
+            options['subtitle'] = self.header.get('subtitle')
+            options['location'] = self.header.get('location')
+            options['date'] = self.header.get('date')
+            options['website'] = self.header.get('website')
+            options['instructions'] = self.header.get('instructions')
+            options['deadline'] = self.header.get('deadline')
+        return options
+
+    def open_submissions(self, due_date = None, subject_areas = [], additional_fields = []):
+
+        ## Author console
+        authors_group = openreview.Group(id = self.id + '/Authors',
+            readers = ['everyone'],
+            signatories = [self.id],
+            signatures = [self.id],
+            writers = [self.id]
+        )
+        self.webfield_builder.set_author_page(self.id, authors_group, { 'title': 'Author console'})
+
+        ## Submission invitation
+        options = {
+            'name': 'Submission',
+            'short_name': self.short_name,
+            'due_date': due_date,
+            'reply': self.type.submission_reply(self.id, subject_areas, additional_fields)
+        }
+        return self.invitation_builder.set_submission_invitation(self.id, options)
+
+    def close_submissions(self, freeze_submissions = True):
+
+        # Get invitation
+        invitation = self.client.get_invitation(id = self.id + '/-/Submission')
+        if invitation.reply['writers'] != self.id:
+            invitation.reply['writers'] = self.id
+
+        now = round(time.time() * 1000)
+        if invitation.duedate > now:
+            invitation.duedate = now
+        self.client.post_invitation(invitation)
+
+        # if freeze submissions then remove writers
```

Comment:
Should check the value of freeze_submissions.

Reply:
You are right! Thanks!